platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java

```diff
@@ -500,6 +543,90 @@ public void testCascadeDelete() {
     }
 
     @Test
+    public void testCascadeDeleteUniDirectional() {
+        ChildCase child1 = new ChildCase("Child 1");
+        childCaseDataService.create(child1);
+
+        ChildCase child2 = new ChildCase("Child 2");
+        childCaseDataService.create(child2);
+
+        MotherCase mother1 = new MotherCase("Mother");
+        mother1.getChildCases().add(child1);
+        mother1.getChildCases().add(child2);
+
+
+        motherCaseDataService.create(mother1);
+
+        MotherCase mother = motherCaseDataService.findById(mother1.getId());
+        assertNotNull(mother);
+
+        ChildCase child = childCaseDataService.findById(child1.getId());
+        assertNotNull(child);
+
+        motherCaseDataService.delete(mother);
+        mother = motherCaseDataService.findById(mother.getId());
+        assertNull(mother);
+
+        child = childCaseDataService.findById(child2.getId());
+        assertNull(child);
+    }
+
+
+    @Test
+    public void testCascadeDeleteMap() {
+
+
+        Event event = new Event("Say Hello");
+        event.getData().put("heading", "Hello Message");
+        event.getData().put("from", "xx@mm.com");
+        eventDataService.create(event);
+
+
+        EventLog loggingSystem = new EventLog("Logging System");
+        loggingSystem.getEvents().add(event);
+
+
+        eventLogDataService.create(loggingSystem);
+
+        EventLog eventLog = eventLogDataService.findById(loggingSystem.getId());
+        assertNotNull(eventLog);
+
+        event = eventDataService.findById(event.getId());
+        assertNotNull(event);
+
+        eventLogDataService.delete(eventLog);
+        eventLog = eventLogDataService.findById(eventLog.getId());
+        assertNull(eventLog);
+
+        event = eventDataService.findById(event.getId());
+        assertNull(event);
+
+        Unit unit = new Unit("Tree");
```

Comment:
Seems like all of the below belongs to a new test. Also not sure why are we testing unidirectional relationships again.